apps/job-core/src/amqp_to_redis.js

```diff
@@ -0,0 +1,76 @@
+import redis from 'redis'
+import { v4 as uuid } from 'uuid'
+import { promisify } from 'util'
+
+const sleep = promisify(setTimeout)
+
+class AMQPToRedis {
+  constructor(redis_url) {
+    this.client = redis.createClient({ url: redis_url })
+    this.lpop = promisify(this.client.lpop).bind(this.client)
+    this.quit = promisify(this.client.quit).bind(this.client)
+    this.closed = false
+    this.ack_callbacks = {}
+    this.exec_message = promisify((content, message_callback, callback) => {
+      const id = uuid()
+      this.ack_callbacks[id] = callback
+      message_callback({id: id, content: content})
+    }).bind(this)
+  }
+
+  queueName(queue) {
+    return 'queue_' + queue
+  }
+
+  ack(message) {
+    const callback = this.ack_callbacks[message.id]
+    if (callback) {
+      this.ack_callbacks[message.id] = null
+      callback()
```

Comment:
Should this be done synchronously?  This will block other messages from being received or processed as the callback is running.  Is this desirable behaviour or am I missing something?

Reply:
Nope, we are duplicating the behaviour of the AMQP API.

Reply:
That's all I need to hear 😊